framework/transport/ieee1905_transport/ieee1905_transport.cpp

```diff
@@ -6,94 +6,91 @@
-            handle_netlink_pollin_event();
-        }
-        if (revents & MAPF_POLLERR) {
-            MAPF_ERR("got MAPF_POLLERR event on netlink socket.");
-            // TODO: handle this
+    // Create a shared_ptr socket wrapper for the netlink socket
+    auto netlink_socket = std::shared_ptr<Socket>(new Socket(netlink_fd_), [](Socket *socket) {
+        // Close the socket file descriptor
+        if (socket) {
+            close(socket->getSocketFd());
         }
+    });
+
+    // Add the netlink socket into the broker's event loop
+    m_broker->add_event(
+        netlink_socket,
+        {
+            // Accept incoming connections
+            .on_read =
+                [&](BrokerServer::EventType socket, BrokerServer::EventLoopType &loop) {
+                    LOG(DEBUG) << "incoming message on the netlink socket";
+                    handle_netlink_pollin_event();
+                    return true;
+                },
+
+            // Not implemented
+            .on_write   = nullptr,
+            .on_timeout = nullptr,
+
+            // Fail on server socket disconnections or errors
+            .on_disconnect =
+                [&](BrokerServer::EventType socket, BrokerServer::EventLoopType &loop) {
+                    LOG(ERROR) << "netlink socket disconnected";
+                    return false;
+                },
+            .on_error =
+                [&](BrokerServer::EventType socket, BrokerServer::EventLoopType &loop) {
+                    LOG(ERROR) << "netlink socket error";
+                    return false;
+                },
+        });
 
-        // check for events on all active network interface sockets
-        for (auto it = network_interfaces_.begin(); it != network_interfaces_.end(); ++it) {
-            auto &network_interface = it->second;
-
-            if (network_interface.fd >= 0 && !network_interface.is_bridge) {
-                MAPF_DBG("check for events on interface " << network_interface.ifname << ".");
-                revents = poller_.CheckEvent(network_interface.fd);
-                if (revents & MAPF_POLLIN) {
-                    MAPF_DBG("got MAPF_POLLIN event on interface " << network_interface.ifname
-                                                                   << ".");
-                    handle_interface_pollin_event(network_interface.fd);
-                }
-                if (revents & MAPF_POLLERR) {
-                    // this could happen whenever an interface comes down
-                    MAPF_DBG("got MAPF_POLLERR event on interface " << network_interface.ifname
-                                                                    << " (disabling it).");
-                    handle_interface_status_change(if_nametoindex(network_interface.ifname.c_str()),
-                                                   false);
-                }
-            }
+    // Run the broker event loop
+    for (;;) {
```

Comment:
I *love* that we don't need to do any cleanup, so no signal handler needed, so we can simply do an endless loop here.